src/com/amazon/ion/impl/LocalSymbolTableImports.java

```diff
@@ -278,6 +278,20 @@ SymbolTable getSystemSymbolTable()
         return myImports;
     }
 
+    /**
+     * Gets the {@link ImportLocation} for the given symbol ID.
+     * @param sid the symbol ID.
+     * @return an ImportLocation, or null if the given symbol ID is not present in the imports.
+     */
+    ImportLocation getImportLocation(int sid) {
+        for (int i = myBaseSids.length - 1; i >= 0; i--) {
+            if (myBaseSids[i] < sid) {
+                return new ImportLocation(myImports[i].getName(), sid - myBaseSids[i]);
+            }
+        }
+        return null;
```

Comment:
If only we could have [nice things](https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#binarySearch(int[],%20int)). Thanks for confirming the context here offline- that `myBaseSids` are sorted, that SIDs in each import are increasing and non-overlapping. 

Not at all essential, but could you drop a Java 1.6+ TODO or something here so that when we finally clear that obstacle we don't forget?

Reply:
+1, added a comment.